rplugin/python3/denite/ui/default.py

```diff
@@ -107,13 +107,28 @@ def init_buffer(self):
         self.__bufvars['denite_statusline_left'] = ''
         self.__bufvars['denite_statusline_right'] = ''
 
-        self.__vim.command('syntax case ignore')
-        self.__vim.command('highlight default link deniteMatched Search')
+        self.__init_syntax()
 
         if self.__context['statusline']:
             self.__window_options['statusline'] = \
                 '%{denite#get_status_left()} %=%{denite#get_status_right()}'
 
+    def __init_syntax(self):
+        self.__vim.command('syntax case ignore')
+        self.__vim.command('highlight default link deniteMatched Function')
+
+        sources = self.__denite.get_sources()
+        has_multiple_sources = len(self.__context['sources']) > 1
+        for raw_source in self.__context['sources']:
+            name = raw_source['name']
+            syntax_line = 'syntax match %s /%s/ nextgroup=%s keepend' % (
```

Comment:
`Base` has a no-op base method `highlight_syntax`. It indicates that every source may have a customized highlight method, thus making the code self-documented. And in source code we can skip a method check.

I can use `hasattr` call. But I wonder what is the benefit, maybe saving memory because base class has fewer methods?